### DISCUSSION TITLE
Panel 1 Q/A

### DISCUSSION THREAD

**Discussion by cmbarton** [cssi-pi-community/2022-meeting#7] [Panelist Q/A for all discussions]
### Post a question for panelists in a comment responding to this discussion thread.

**Comment by pibion** [comment]
I cannot agree strongly enough on the importance for maintenance funding.  In my experience tests, documentation, and a responsive team make a huge difference for the approachability of a library for students.  Are there NSF pathways for this?

**Reply by rapier1** [reply]
I was at PI meeting several years ago where we PIs stressed over and over again that maintenance funding was critical. We were told by the program officers in attendance that there were no plans to provide maintenance funding. Instead, it was up to the PI to develop long term funding through the participation of their host institution, community involvement, and so forth. I really do hope that the NSF is rethinking this position.

**Reply by pibion** [reply]
Maintenance funding has been a huge topic of discussion at SNOWMASS as a critical community need in the Cosmic Frontier.  

Are there successful examples of projects that are well-maintained?  How can we tackle this issue??

**Reply by rapier1** [reply]
I've been maintaining HPN-SSH for 18 years. It's always been a matter of me finding the time to work on it. The Elements grant I received for HPN-SSH in 2020 is kind of a maintenance grant but it definitely required novel engineering and research components as well. NB: HPN-SSH is a one person show for the most part so my experiences aren't really applicable to larger projects.

**Reply by sunbiz** [reply]
The POSE call might be addressing this, but it definitely needs more funding - https://www.nsf.gov/pubs/2022/nsf22572/nsf22572.htm and better funding mechanisms

**Reply by pibion** [reply]
I think it's the "novel" part that I find problematic.  If there could be funding for projects that have a clear need but aren't necessarily novel that would help a lot.  For example, if you could propose documentation improvements because there's evidence the user base is struggling with using certain functions, or propose maintenance that helps improve sustainability, or propose maintenance that improves interoperability.

**Reply by rapier1** [reply]
I agree that there is a problem in those terms. Having a stable funding stream for important software projects is, in my view, critical. This is especially true for projects where the team members are all on soft money. Without a grant many of us simply can't justify the time that might be required to maintain valuable resources. We can ask the host institution for money/resources/time but getting that charge string can be nearly impossible.

**Reply by munkm** [reply]
@pibion I think that both numpy and matplotlib have been successful getting funding for maintenance, though that funding also came with new feature development. I will try to find links of their proposals.

**Reply by munkm** [reply]
These aren't the proposals that they submitted, but it does include where funding is coming from for these open source projects

Here is matplotlib's [current list](https://matplotlib.org/governance/funding.html#current). I think their NASA funding is from the ROSES call. 

Here is the [NEP for numpy](https://numpy.org/neps/nep-0048-spending-project-funds.html#numpy-funding-history-and-current-status) describing their current funding and how money is dispersed.

**Comment by pibion** [comment]
Most of the people in my lab are undergraduates and there are very few RSE positions that don't expect a PhD.  Most of my students go into industry, and one has come back from industry into a PhD program through the https://www.nsf.gov/cise/CSGrad4US/ program.

But there aren't many (any?) programs that are a clear pathway for students who are interested in cyberinfrastructure/software infrastructure.  Is there a role for NSF in helping this part of our ecosystem?  What can we do about students who don't excel in traditional academic environments and don't have a grad program to go to?

**Reply by carolxsong** [reply]
Some institutions have CI apprentice-style programs to train and bring up undergraduate students into the research computing/research software organizations. NSF's cybertraining's CIP track may be a good mechanism for what you are looking for.

**Reply by reedmaxwell** [reply]
Good question and an important one.  [Princeton has a terrific RSE program](https://researchcomputing.princeton.edu/services/research-software-engineering) that is a split approach.  RSE's sit in the RSE department (in research computing) and have career advancement and some permanence, and are embedded in the disciplinary research groups.  I think this is a great model, my $0.02.

**Reply by pibion** [reply]
That looks super cool!  My problem is less about getting undergrad researchers and more about where to send them to.  They like the flexibility and motivation of academic research, but often can't afford to say no to the money and health care available in industry.  Also, some of them aren't a good fit for grad programs.

**Reply by reedmaxwell** [reply]
@pibion ahh, yes, the RSE's at Princeton don't all have PhDs, many are BS or MS.  This allows a much more flexible (broad) array of RSE jobs to be represented (support and research) and allows for mentorship and advancement.

**Reply by mbrazil1** [reply]
I like the model of having a pool of RSEs that can embed with specific research groups. We are hoping to work toward something like this at ASU with our CI facilitators/RSEs. @pibion thanks for elevating the question about students!

**Comment by pibion** [comment]
What are your thoughts about the need for academic departments focused on cross-cutting computing and software needs?

Are there institutions that have built stable jobs for research software engineers?  Is this purely an institutional issue?  Or is this something that funding agencies have input on in any way?

**Reply by mbrazil1** [reply]
I do think that a few institutions have successfully created stable roles for RSEs. At ASU, we have two RSEs / Facilitators that help researchers on various projects. We have a built in mechanism to provide consulting services to research groups (for a fee) if the project goes beyond a certain number of hours. Right now, most projects are below that threshold, though.

**Reply by munkm** [reply]
I think academic departments focused on cross-cutting computing and software needs are totally necessary for transforming science! There are so many tools and methods that are used across domains, and we could benefit more from sharing these tools. More broadly, I also think these departments would be a great home for RSEs, especially if the department is focused on research software development and use. It may be worth reaching out to the US research software engineer association to think about how we'd like to see RSE positions evolve in the US https://us-rse.org/ . Interestingly, the [job board](https://us-rse.org/jobs/) on their page does show RSE positions at universities and at national laboratories.

**Comment by terletska** [comment]
What are your thoughts about increasing diversity in terms of universities and/or states-wise: on expanding funding towards the universities which don't have much CSSI funding (e.g., R2 schools)? Would it be helpful to encourage partnership on proposals with such schools to help them get started?
This is related to the rich getting richer problem.

**Reply by dtarb** [reply]
I wish there were not school categorizations (ivy league, R1, R2 etc).  I happen to be at a university that has worked hard and claims pride in just having achieved R1.  Now this may limit my institution from opportunities it previously had - perhaps an unintended consequence of trying to elevate itself, which may or may not have been a worthwhile effort in the first place and was driven by motivations to attract better students, staff and funding.  In my opinion it would be better not to have categorizations and leave it for universities to establish their own credibility and reputation in all areas including diversity.  If it is CSSI funding that needs to be more equitably spread - should it not be targeted at universities that do not already have CSSI - regardless of R level, i.e. the direct concern, rather than R level as a surrogate.

**Comment by lums658** [comment]
Following up on Ritu's comments about enhancing a culture vs fitting into a culture.  I think it is essential in diversity, equity, and inclusion discussions to emphasize that diversity is a strength, not a chore.  Building an effective team is not accomplished through hiring the most previously successful individuals. There is an interesting paper by Wooley, et al., " Evidence for a Collective Intelligence Factor in the Performance of Human Groups," that showed that the factors that correlated most highly with team performance were average social sensitivity of the group, equality of conversational turn taking, and the number of women in the group -- *not*, say, the maximum individual IQ or even average IQ of the group.

**Reply by gkthiruvathukal** [reply]
Spot on, @lums658 and Ritu (not sure of your GitHub ID). Hope to see you during one of the breaks. I think your point on Collective Intelligence is well grounded in Gardner's longstanding theory of [multiple intelligences](https://www.simplypsychology.org/multiple-intelligences.html).  Because we are not born with all of the intelligence we will have--or need--in our lifetime, we need to build teams from diverse backgrounds and experiences and not discount that we have the potential within us to get smarter -- both intellectually and emotionally -- as we progress through various life stages. In this vein, I'd also like to see more [Emotional Intelligence](https://www.danielgoleman.info/) in discussions about culture, which I think would greatly improve group dynamics and collaboration.

**Comment by manning-ncsa** [comment]
This is a "meta" comment: It was not obvious that this discussion forum was publicly visible. If this was not explicitly mentioned in the introduction to the tool, I encourage the organizers to do so.

**Reply by jlmays** [reply]
This was a technical glitch that was just noticed now. Sorry

**Comment by LSSchadler** [comment]
Duke University has a pool of software developers that investigators can hire as their grants come in.  This ensures job stability for the developers and provides expertise to help domain experts trying to develop software infrastructure.   What if NSF had a pool of developers that could be part of a grant?

**Reply by lagc03** [reply]
As a software developer for a CSSI project I can say that the model at your institution is something all institutions must seek (maybe through their office of research projects). Most projects require websites, database setups and computing environments that can be easily standardized, leaving more time for the actual research. NSF may help by pushing institutions to do just that via funding to launch the initiative and later on as a requirement for funding in projects that need CI.

**Reply by pibion** [reply]
@LSSchadler this sounds a little like the SGCI model.  SGCI used to be free, but now I think you have to write them into your grant.

**Comment by erjank** [comment]
What efforts go into selecting panelists who can provide expert feedback on efforts to diversify and make more equitable the training of CI professionals that are included in CSSI proposals, or is it the case that we are relying on the opinions of great CI researchers who may not be experienced or invested in dismantling systems of oppression?

**Reply by cmbarton** [reply]
> What efforts go into selecting panelists who can provide expert feedback on efforts to diversify and make more equitable the training of CI professionals that are included in CSSI proposals, or is it the case that we are relying on the opinions of great CI researchers who may not be experienced or invested in dismantling systems of oppression?

The organizing committee and NSF proactively worked to find panelists who would have the expertise, professional  experience, and interest to provide useful insights on each of the three panel topics.

**Reply by cmbarton** [reply]
Many/most panelists are **not** CSSI PIs, although some are

**Reply by erjank** [reply]
Thank you!

**Comment by kshalm** [comment]
I like the comment about cognitive diversity. I think this can apply to proposal funding where reviewers can rate proposals higher that fit with their world views. Everyone here has benefited from this funding model (we were funded, after all). What about using random sampling to find some fraction of proposals? Make sure the proposals meet some minimum quality checks, and then randomly select which ones get funded.  This might fund impactful projects that would traditionally get overlooked.

**Reply by pibion** [reply]
Ooooh this is a fascinating idea!!

**Comment by erjank** [comment]
I often hear "I'm not a social scientist so..." as a bit of an abdication of responsibility for effecting cultural change, but maybe we can push back at that? We all learned how to be scientists in our own disciplines, maybe a bit more learning isn't so out of reach?

**Reply by pibion** [reply]
@erjank I'm definitely on board for personal responsibility.  For me, wanting to collaborate with social scientists is more about feeling like their expertise could help effect larger-scale changes.  The TEAM-UP report is incredibly useful, and I don't know if physicists alone could have written that.

A lot of my view about what might work is based on personal or near-personal experience.  And it's often at odds with pipeline statistics, at least in physics.  But my sample size is so small!  I don't know that my experiences are necessarily so universal.

I do think my experiences, plus a bit of reading, can be useful for making my lab a good place to work, at least for the issues I'm aware of.  I also think pairing my experience and data savvy with social scientists who specialize in science retention would make for better community-wide recommendations.

**Reply by cmbarton** [reply]
> I often hear "I'm not a social scientist so..." 

We can also include more social scientists in interdisciplinary research teams. Diverse, interdisciplinary teams are especially important in programs like CSSI

**Reply by munkm** [reply]
> We can also include more social scientists in interdisciplinary research teams. 

Yes. I think social scientists have training on/in different skills than people in my field. I would love to collaborate with social scientists on studies like what was suggested in the panel, but I don't have the domain expertise to do it on my own. I think bringing in a wholistic approach with these larger problems, like diversity and inclusion, by including the expertise of social scientists and ethnographers would be beneficial for transforming the field.

**Reply by cmbarton** [reply]
And there is a rapidly growing domain of computational social science

**Comment by terletska** [comment]
Excellent comment by M. Sokoloff on the "elephant in a room" that the pipeline leak starts from not getting students into the pipeline. One of the common responses of students is that "CI is not for women, it is for men". The lack of role models is one of the biggest factors making students believe this.

**Reply by pibion** [reply]
Physicist here, I want to second the need for investing in good social science.

My experience is that students leave despite great interest - but that's just my experience!  Something like the TEAM-UP report for this field would be amazing.

**Reply by pibion** [reply]
I've changed my intro physics classes over to emphasize numerical computation.  This makes their problem solving easier to debug and gets everyone involved in computation.

It's been a good recruitment tool for my research and it's a lot of fun, we can solve more real-world problems with numerical methods.

**Reply by kshalm** [reply]
The pipeline problem is complex. It isn’t just an interest problem, but also an accessibility problem.

**Comment by kshalm** [comment]
Table 3 Discussion:

1) Is the level of cultural diversity in science or in universities different than the real world? Can cultural diversity from different countries mask other inequities? 

2) Expanding on the randomized funding comment above, could such a system help increase institutional diversity and  cognitive diversity? Also, could this lead to an increase in the number of proposals from underrepresented groups or institutions that might otherwise self select themselves out of submitting?

3) A solution for software project maintenance is a critic question that needs to be answered.
